tests/kernel/spinlock/src/utils.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2018 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+
+#define EFLAGS_IF       BIT(9)
+
+/* Macro to save rflags register */
+#define CPU_RFLAGS_SAVE(rflags_ptr)			\
+{							\
+	__asm__ volatile (" pushf");			\
+	__asm__ volatile (" pop %0"			\
+			: "=r" (*(rflags_ptr))		\
+			: /* No inputs */);		\
+}
+
+/* This helper function use to check if interrput enable or not */
+bool check_interrupt_enable_or_not(void)
+{
+#if defined(CONFIG_X86)
+	ulong_t eflag;
+
+	CPU_RFLAGS_SAVE(&eflag);
+
+	return (bool)(eflag & EFLAGS_IF);
+#else
+	int irq_mask = 255;
+	int irq_enabled;
+
+	irq_enabled = irq_is_enabled(irq_mask);
```

Comment:
This isn't the correct API to use.  irq_is_enabled() tells you whether or not specific interrupt lines are enabled.  It's for use by drivers to control masking of their own IRQ while they do stuff like initialization.  It does not tell you whether or not interrupts are currently globally unmasked in the sense of EFLAGS (x86), PRIMASK/BASEPRI (arm), PS.INTLEVEL (xtensa), et. al.

We actually don't have that API in a global sense.  What we do have is `arch_irq_unlocked(key)`, which takes as an **argument** the key returned form arch_irq_lock() and tells you whether, when you call arch_irq_unlock() with that key, interrupts will be globally unmasked (it was designed to be able to tell when we're inside a nested irq lock).  That's what you want to use to do this portably.

Reply:
Appreciate for giving this clever way. Do it this way avoiding writing the arch-dependent code, and it still can check  interrupt mask happened. I have deleted the unnecessary code (utils.h) and update again.